forbiddenfruit/__init__.py

```diff
@@ -218,25 +219,10 @@ class PyAsyncMethods(ctypes.Structure):
 }
 
 
-class SlotsProxy(PyObject):
-    _fields_ = [('dict', ctypes.POINTER(PyObject))]
-
-
 def patchable_builtin(klass):
-    name = klass.__name__
-    target = klass.__dict__
-
-    proxy_dict = SlotsProxy.from_address(id(target))
-    namespace = {}
-
-    # This code casts `proxy_dict.dict` into a python object and
-    # `from_address()` returns `py_object`
-    ctypes.pythonapi.PyDict_SetItem(
-        ctypes.py_object(namespace),
-        ctypes.py_object(name),
-        proxy_dict.dict,
-    )
-    return namespace[name]
+    refs = gc.get_referents(klass.__dict__)
+    assert len(refs) == 1
```

Comment:
Do you think something like this would be useful at all?
I didn't dig too much in how `get_referents` can fail, but thought it'd be nice to leave a lil clue in case it does:
```suggestion
    assert len(refs) == 1, "Failed retrieving class __dict__"
```

Reply:
It can't fail (well, only failure reason is lack of memory, in which case it'll raise an exception anyway). The aim of this check is to ensure there is indeed only 1 reference from `dictproxy`/`mappingproxy`, because in current CPython versions, they reference exactly one object. If at any point in the future they'll have additional GC-tracked fields, then the result list will be longer, and we will need to fix the code to account for that.

I think it's less important to add a clue in this case, because it's not really a "bug" that can happen in runtime for users, it can only happen when we test forbiddenfruit on (hypothetical) new CPython versions (and if it happens, we'll have to look and check the order of items in the new list returned; and possibly always take the first item, or the second item, depending on the order of fields in the object)

Reply:
I see. thank you for the thorough explanation.